ush/detect_machine.sh

```diff
@@ -58,16 +58,19 @@ fi
 # Overwrite auto-detect with MACHINE if set
 MACHINE_ID=${MACHINE:-${MACHINE_ID}}
 
+# Overwirte auto-detect of MACHINE_ID if in container
+if [[ -d /opt/spack-stack ]]; then
+  # We are in a container
+  MACHINE_ID=container
+fi
+
```

Comment:
This is probably fine for the machines we support presently, but you may want to find another way to identify containers since `/opt` is a common installation location for software in the Linux world.  I'm not sure, but I would guess that singularity or apptainer would set an environment variable when working in a container.